ce/marketing/developer/extend-segments.md

```diff
@@ -211,104 +211,72 @@ With the delete request, you delete the compound segment that is created.
 }
 ```
 
-**Remove a segment member**
+**Remove segment members by providing ids**
 
 ```HTTP
 POST {{OrgUrl}}/api/data/v9.0/msdyncrm_SegmentMembersUpdate
 {
-    "msdyncrm_segmentid": "59AC8BBF-57E7-E811-A9A9-000D3A35F403",
-    "msdyncrm_operation": "removeByIds",
-    "msdyncrm_memberids": "[\"B5672BDB-8899-43CB-9FA1-0AE4DC61DAD3\"]",
-    "msdyncrm_segmentquery": null
+   "msdyncrm_segmentid": "59AC8BBF-57E7-E811-A9A9-000D3A35F403",
+   "msdyncrm_operation": "removeByIds",
+   "msdyncrm_memberids": "[\"B5672BDB-8899-43CB-9FA1-0AE4DC61DAD3\", \"694E1C8E-F704-4B23-9B07-E65DB1620E47\", \"A4A31E3D-DFCA-4765-8018-3BA7D5E376C7\"]"
+}
 ```
 
-**Remove multiple segment members**
+**Add segment members by providing query**
 
 ```HTTP
 POST {{OrgUrl}}/api/data/v9.0/msdyncrm_SegmentMembersUpdate
 {
-   "msdyncrm_segmentid": "59AC8BBF-57E7-E811-A9A9-000D3A35F403",
-   "msdyncrm_operation": "removeByIds",
-   "msdyncrm_memberids": "[\"B5672BDB-8899-43CB-9FA1-0AE4DC61DAD3\", \"694E1C8E-F704-4B23-9B07-E65DB1620E47\", \"A4A31E3D-DFCA-4765-8018-3BA7D5E376C7\"]",
-   "msdyncrm_segmentquery": null
+    "msdyncrm_segmentid": "b5466fbb-2cef-e911-a81d-000d3a6d200c",
+    "msdyncrm_operation": "addByQuery",
+    "msdyncrm_query": "PROFILE(account, account_1).FILTER(account_1.accountid == '1cc00a15-37ef-e911-a81d-000d3a6d200c').TRAVERSE(contact_account_parentcustomerid, contact_1).FILTER(ISNOTNULL(contact_1.emailaddress1))"
 }
 ```
 
-**Add segment members by query**
+**Remove segment members by providing query**
 
 ```HTTP
 POST {{OrgUrl}}/api/data/v9.0/msdyncrm_SegmentMembersUpdate
 {
     "msdyncrm_segmentid": "b5466fbb-2cef-e911-a81d-000d3a6d200c",
-    "msdyncrm_operation": "addByQuery",
-    "msdyncrm_memberids": null,
-    "msdyncrm_segmentquery": "PROFILE(account, account_1).FILTER(account_1.accountid == '1cc00a15-37ef-e911-a81d-000d3a6d200c').TRAVERSE(contact_account_parentcustomerid, contact_1).FILTER(ISNOTNULL(contact_1.emailaddress1))"
+    "msdyncrm_operation": "removeByQuery",
+    "msdyncrm_query": "PROFILE(account, account_1).FILTER(account_1.accountid == '1cc00a15-37ef-e911-a81d-000d3a6d200c').TRAVERSE(contact_account_parentcustomerid, contact_1).FILTER(ISNOTNULL(contact_1.emailaddress1))"
 }
 ```
 
-**Remove segment members by query**
+**Get status of pending operations**
 
 ```HTTP
 POST {{OrgUrl}}/api/data/v9.0/msdyncrm_SegmentMembersUpdate
 {
-    "msdyncrm_segmentid": "b5466fbb-2cef-e911-a81d-000d3a6d200c",
-    "msdyncrm_operation": "removeByQuery",
-    "msdyncrm_memberids": null,
-    "msdyncrm_segmentquery": "PROFILE(account, account_1).FILTER(account_1.accountid == '1cc00a15-37ef-e911-a81d-000d3a6d200c').TRAVERSE(contact_account_parentcustomerid, contact_1).FILTER(ISNOTNULL(contact_1.emailaddress1))"
+    "msdyncrm_segmentid":"b5466fbb-2cef-e911-a81d-000d3a6d200c",
+    "msdyncrm_operation":"getState"
 }
 ```
 
-**Retrieve segment members**
+**Retrieve segment members (legacy)**
```

Comment:
I believe the word we usually use in official docs is "obsolete" or "deprecated", not "legacy". I would also put the current, recommended way first, to make it clearer what the preferred way is.